src/components/search/SearchHistory.tsx

```diff
@@ -113,31 +113,43 @@ const SearchHistoryItem = ({ text }: SearchHistoryItemProps) => {
 }
 
 const SearchHistory = () => {
-  const { searchHistory, clearHistory } = useSearchHistory()
+  const { searchHistory, clearHistory, hasFetched } = useSearchHistory()
+  const [displayHistory, setDisplayHistory] = useState(searchHistory)
+
+  // Only update history on component load and clearing of search history
+  useEffect(() => {
+    if (hasFetched) setDisplayHistory(searchHistory)
+    // @ts-ignore do not reset on searchHistory update.
+  }, [hasFetched, setDisplayHistory])
```

Comment:
could be error prone. would a ref make more sense?

Reply:
The search results that are displayed are only updated when the history has been fetch and the "clear" button is hit. 
Otherwise, the component is unmounted and re-fetches the history on component load. 
I don't think a ref would cause a re-render of the search history when hitting the clear btn.

Reply:
you could have both a state var and a ref

Reply:
Are you suggesting that the update flow stay the same w/ this use effect and update in onClearHistory, just changing the `displayHistory` to be a ref instead of state?

Reply:
searchHistory so that you don't have to ts ignore that's all